libraries/chain/hotstuff/hotstuff.cpp

```diff
@@ -55,15 +56,17 @@ pending_quorum_certificate::pending_quorum_certificate(size_t num_finalizers, ui
 
 bool pending_quorum_certificate::is_quorum_met() const {
    std::lock_guard g(*_mtx);
-   return _state == state_t::weak_achieved || _state == state_t::weak_final || _state == state_t::strong;
+   return is_quorum_met_no_lock();
 }
 
 // called by add_vote, already protected by mutex
 vote_status pending_quorum_certificate::add_strong_vote(const std::vector<uint8_t>& proposal_digest, size_t index,
                                                         const bls_public_key& pubkey, const bls_signature& sig,
                                                         uint64_t weight) {
-   if (auto s = _strong_votes.add_vote(proposal_digest, index, pubkey, sig); s != vote_status::success)
+   if (auto s = _strong_votes.add_vote(proposal_digest, index, pubkey, sig); s != vote_status::success) {
+      dlog("add_strong_vote returned failure");
```

Comment:
why is this `dlog` useful since we have a non-conditional `dlog` in `add_vote`?

Reply:
This is only happening when duplicate occurs; useful for debugging.